### PR TITLE
fix: Update AppSettings obj to match new format set by new runtime

### LINKED ISSUE #6647: allowedCaller add in wrong place.
## Describe the bug
When I use new version runtime bot, I add the allowedCaller in settings page, but it was added in wrong place. It need to be added into runtimeSettings if the bot is new version bot.

image
image

## Browser

- [ ] Electron distribution
- [x] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

### PR DESCRIPTION
**Pull request by pavolum**
## Description

New runtime anticipates AllowedCallers within appsettings JSON to  be in runtimeSettings as opposed to within old runtimes skillsConfiguration obj.

This PR:
- Updates appSettings type and defaults to match what is expected by the new runtime (i.e. adding runtimeSettings obj)
- Updates AllowedCaller UI to interface with new location of allowed caller
- Adding AllowedCaller UI to settings UI tab view

## Task Item

fixes #6647

### CODE REVIEW

File: Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterModal.tsx [hunk 1]
```diff
@@ -15,17 +15,12 @@ import { JSONSchema7 } from '@botframework-composer/types';
 import { EditorExtension, PluginConfig } from '@bfc/extension-client';
 import mapValues from 'lodash/mapValues';
 import { JSONSchema7Type } from 'json-schema';
+import { AdapterRecord } from '@botframework-composer/types/src';
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterModal.tsx:18]: This should not import from the src of the types package.

File: Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx [hunk 2]
```diff
@@ -14,12 +14,13 @@ import { IconButton } from 'office-ui-fabric-react/lib/Button';
 import { TooltipHost, DirectionalHint } from 'office-ui-fabric-react/lib/Tooltip';
 import { SharedColors } from '@uifabric/fluent-theme';
 import { JSONSchema7 } from '@botframework-composer/types';
+import { AdapterRecord } from '@botframework-composer/types/src';
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx:17]: Don't import from src.

File: Composer/packages/types/src/settings.ts [hunk 3]
```diff
@@ -67,6 +76,34 @@ export type DialogSetting = {
   botId?: string;
   skillHostEndpoint?: string;
   customFunctions: string[];
+  defaultLocale?: string;
+  runtimeSettings?: {
```
Comment (a-b-r-o-w-n) [inline Composer/packages/types/src/settings.ts:80]: Extract this to its own type.

File: Composer/packages/types/src/settings.ts [hunk 4]
```diff
@@ -67,6 +76,34 @@ export type DialogSetting = {
...
+    features?: {
+      removeRecipientMentions?: boolean;
+      showTyping?: boolean;
+      traceTranscript?: boolean;
+      useInspection?: boolean;
+      blobTranscript?: {
+        connectionString?: string;
+        containerName?: string;
+      };
+      setSpeak: {
```
Comment (a-b-r-o-w-n) [inline Composer/packages/types/src/settings.ts:91]: `setSpeak` is not optional, but all of the properties in it are?

**Review by a-b-r-o-w-n — CHANGES_REQUESTED** [review]
So what happens for existing bots with the current settings object?

File: Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterModal.tsx (hunk 1 above)
Comment (pavolum) [reply Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterModal.tsx:18]: Handled

File: Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx (hunk 2 above)
Comment (pavolum) [reply Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx:17]: Handled

File: Composer/packages/types/src/settings.ts (hunk 4 above)
Comment (pavolum) [reply Composer/packages/types/src/settings.ts:91]: Handled

File: Composer/packages/types/src/settings.ts (hunk 3 above)
Comment (pavolum) [reply Composer/packages/types/src/settings.ts:80]: handled

**Comment by pavolum** [comment]
> So what happens for existing bots with the current settings object?

Thats a good question. My assumption is that the migration flow will ensure that the expected appsettings for all bots that hit this point in the flow are normalized to the new version. I will double check on this. @benbrown is my assumption correct?

**Comment by benbrown** [comment]
Yes this is handled in the migration:

https://github.com/microsoft/BotFramework-Composer/pull/6526/files#diff-af4de9db9846a35f2320310fab8e38d50f1a138cb582f583a1a1716c48656836R508

**Review by a-b-r-o-w-n — DISMISSED** [review]

**Review by a-b-r-o-w-n — APPROVED** [review]

**Review by a-b-r-o-w-n — APPROVED** [review]

**Pull request merged**